src/io/flutter/preview/PreviewView.java

```diff
@@ -182,8 +173,23 @@ public void onResponse(FlutterOutline widget, JsonObject response) {
     }
 
     @Override
-    public void onFailure(Throwable e) {
-      noWidget();
+    public void onFailure(RenderProblemKind kind) {
+      ApplicationManager.getApplication().invokeLater(() -> {
+        if (previewArea != null) {
+          switch (kind) {
+            case EXCEPTION:
+            case NO_TEMPORARY_DIRECTORY:
+              previewArea.clear("There was an exception during rendering");
+              break;
+            case NO_WIDGET:
+              previewArea.clear("The selection does not correspond to a widget");
```

Comment:
Will this also show when the widget is missing a `forDesignTime()` constructor?

Reply:
It might be good to distinguish between widgets in general and un-renderable widgets.  Maybe as simple as changing the text to:

    "The selection does not correspond to a renderable widget"

In a future release we could add a link to take users to docs explaining what this means...

Reply:
Yes, right now we cannot distinguish "no widget" and "there is a widget, but it does not have `forDesignTime()` constructor". I will enhance the Flutter Outline API to fix this.

Reply:
Great.  Thanks!